App/Controllers/Ajax/Category.php

```diff
@@ -0,0 +1,131 @@
+<?php
+
+namespace App\Controllers\Ajax;
+
+use App\Models\CategoryModel;
+use Core\AjaxController;
+use Core\Container;
+use Core\Traits\StringFunctions;
+
+class Category extends AjaxController
+{
+    use StringFunctions;
+
+    protected $slug;
+
+    public function __construct(Container $container)
+    {
+        $this->loadModules[] = 'Slug';
+        parent::__construct($container);
+    }
+
+    /**
+     * Create a new category via Ajax
+     * @throws \Core\JsonException
+     */
+    public function new()
+    {
+        //security checks
+        $this->onlyAdmin();
+        $this->onlyPost();
+
+        //preparing our return results
+        $result = array();
+        $categoryUpdateJson = ($this->request->getData('category-new'));
+        $categoryUpdate = json_decode($categoryUpdateJson);
+
+        //Converting our array of objects to simple array
+        $send = array();
+        foreach ($categoryUpdate as $item) {
+            $send[$item->name] = $item->value;
+        }
+
+        if(!$this->slug->isSlugValid($send["categories_slug"]))
+        {
+            $result["success"] = false;
+            $result["errorMessage"] = "Invalid Slug";
+            echo json_encode($result);
+            die();
+        }
+
+        $categoryModel = new CategoryModel($this->container);
```

Comment:
inject in your constructor

Reply:
Resolved in branch Security_and_bug_fixes